packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -154,6 +149,231 @@ export interface NftDetectionConfig extends BaseConfig {
   selectedAddress: string;
 }
 
+export type ReservoirResponse = {
+  tokens: TokensResponse[];
+  continuation?: string;
+};
+
+export type TokensResponse = {
+  token: TokenResponse;
+  ownership: Ownership;
+  market?: Market;
+  blockaidResult?: Blockaid;
+};
+
+export enum BlockaidResultType {
+  Benign = 'Benign',
+  Spam = 'Spam',
+  Warning = 'Warning',
+  Malicious = 'Malicious',
+}
+
+export type Blockaid = {
+  contract: string;
+  chainId: number;
+  result_type: BlockaidResultType;
+  malicious_score: string;
+  attack_types: object;
+};
+
+export type Market = {
+  floorAsk?: FloorAsk;
+  topBid?: TopBid;
+};
+
+export type TokenResponse = {
+  chainId: number;
+  contract: string;
+  tokenId: string;
+  /** @description Can be erc721, erc115, etc. */
+  kind?: string;
+  name?: string;
+  image?: string;
+  imageSmall?: string;
+  imageLarge?: string;
+  metadata?: Metadata;
+  description?: string;
+  /** @description Can be higher than one if erc1155. */
+  supply?: number;
+  remainingSupply?: number;
+  /** @description No rarity for collections over 100k */
+  rarityScore?: number;
+  rarity?: number;
+  /** @description No rarity rank for collections over 100k */
+  rarityRank?: number;
+  media?: string;
+  /** @default false */
+  isFlagged?: boolean;
+  /** @default false */
+  isSpam?: boolean;
+  /** @default false */
+  isNsfw?: boolean;
+  /** @default false */
+  metadataDisabled?: boolean;
+  lastFlagUpdate?: string;
+  lastFlagChange?: string;
+  collection?: Collection;
+  lastSale?: LastSale;
+  topBid?: TopBid;
+  /** @description The value of the last sale.Can be null. */
+  lastAppraisalValue?: number;
+  attributes?: Attributes[];
+};
+
+export type TopBid = {
+  id?: string;
+  price?: Price;
+  source?: {
+    id?: string;
+    domain?: string;
+    name?: string;
+    icon?: string;
+    url?: string;
+  };
+};
+
+export type LastSale = {
+  /** @description Deprecated. Use `saleId` instead. */
+  id?: string;
+  /** @description Unique identifier made from txn hash, price, etc. */
+  saleId?: string;
+  token?: {
+    contract?: string;
+    tokenId?: string;
+    name?: string;
+    image?: string;
+    collection?: {
+      id?: string;
+      name?: string;
+    };
+  };
+  orderSource?: string;
+  /**
+   * @description Can be `ask` or `bid`.
+   * @enum {string}
+   */
```

Comment:
We're in TypeScript-land, so we don't need to describe what the type of this property is, as it can be picked up automatically:

``` suggestion
```